sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ContextualServer.java

```diff
@@ -79,57 +76,49 @@ private void startSonarTSServer() {
 
     try {
       processBuilder.inheritIO();
-      serverProcess.set(processBuilder.start());
-      // TODO find a better way to wait until the server is up
-      // e.g. loop with accessing the port
-      synchronized (this) {
-        wait(1000);
-      }
+      serverProcess = processBuilder.start();
+      tryToConnect();
       LOG.info("SonarTS Server is started");
-
     } catch (IOException e) {
       LOG.error("Failed to start SonarTS Server", e);
-
-    } catch (InterruptedException e) {
-      LOG.error("Failed to wait until the SonarTS Server is up", e);
-      throw new RuntimeException(e);
     }
   }
 
+  private static void tryToConnect() throws IOException {
+    InetSocketAddress address = new InetSocketAddress("localhost", 55555);
+    Socket socket = new Socket();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use try-with-resources or close this "Socket" in a "finally" clause. ([squid:S2095](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2095))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=issue-446&issues=AWEDyrk3f6JiG_HArpQ8&open=AWEDyrk3f6JiG_HArpQ8)

[ISSUE_KEY]: AWEDyrk3f6JiG_HArpQ8